eslint/eslint-patch/src/eslint-bulk-suppressions/bulk-suppressions-patch.ts

```diff
@@ -198,33 +217,92 @@ export function write(): void {
   }
 }
 
-// utility function for linter-patch.js to make require statements that use relative paths in linter.js work in linter-patch.js
-export function requireFromPathToLinterJS(importPath: string): import('eslint').Linter {
-  if (!eslintFolder) {
-    return require(importPath);
+function binarySearch(arr: number[], target: number, low: number, high: number): number {
+  while (low <= high) {
+    // eslint-disable-next-line no-bitwise
+    const mid: number = (low + high) >> 1;
+    const midVal: number = arr[mid];
+
+    if (midVal === target) {
+      return mid;
+    } else if (midVal < target) {
+      low = mid + 1;
+    } else {
+      high = mid - 1;
+    }
   }
 
-  const pathToLinterFolder: string = `${eslintFolder}/lib/linter`;
-  const moduleAbsolutePath: string = require.resolve(importPath, { paths: [pathToLinterFolder] });
-  return require(moduleAbsolutePath);
+  // eslint-disable-next-line no-bitwise
+  return ~low;
 }
 
-export function patchClass<T, U extends T>(originalClass: new () => T, patchedClass: new () => U): void {
-  // Get all the property names of the patched class prototype
-  const patchedProperties: string[] = Object.getOwnPropertyNames(patchedClass.prototype);
-
-  // Loop through all the properties
-  for (const prop of patchedProperties) {
-    // Override the property in the original class
-    originalClass.prototype[prop] = patchedClass.prototype[prop];
+function getScopeIdMap(
+  traverser: ITraverser,
+  sourceCode: ISourceCode,
+  positions: number[]
+): Map<number, string> {
+  const scopeIdMap: Map<number, string> = new Map();
+  if (positions.length === 0) {
+    return scopeIdMap;
   }
 
-  // Handle getters and setters
-  for (const [prop, descriptor] of Object.entries(Object.getOwnPropertyDescriptors(patchedClass.prototype))) {
-    if (descriptor.get || descriptor.set) {
-      Object.defineProperty(originalClass.prototype, prop, descriptor);
+  let low: number = 0;
+  let high: number = positions.length - 1;
+
+  const max: number = positions[high];
+
+  const highStack: number[] = [];
+
+  traverser.traverse(sourceCode.ast, {
```

Comment:
This is producing suboptimal results when the desired position is inside of a comment.

Reply:
Reverting fixing of nodes for the moment.